### ISSUE TITLE
IndexOutOfRangeException

### ISSUE THREAD

**Issue by lars-w-pedersen** [Geta/geta-notfoundhandler#3]
i have installed Geta.NotFoundHandler.Optimizely 1.0.0-rc
in startup i have added:

in ConfigureServices:
var connectionString = _configuration.GetConnectionString("EPiServerDB");
services.AddNotFoundHandler(o => {
    o.UseSqlServer(connectionString);
});
services.AddOptimizelyNotFoundHandler();

in Configure:
app.UseNotFoundHandler();

When running i get the following error when the app.UseNotFoundHandler() line is reached:

System.IndexOutOfRangeException
  HResult=0x80131508
  Message=Cannot find table 0.
  Source=System.Data.Common
  StackTrace:
   at System.Data.DataTableCollection.get_Item(Int32 index)
   at Geta.NotFoundHandler.Data.SqlDataExecutor.ExecuteQuery(String sqlCommand, IDbDataParameter[] parameters)
   at Geta.NotFoundHandler.Data.SqlRedirectRepository.GetAll()
   at Geta.NotFoundHandler.Core.Redirects.DefaultRedirectsService.GetAll()
   at Geta.NotFoundHandler.Core.Redirects.RedirectsInitializer.GetCustomRedirects()
   at Geta.NotFoundHandler.Core.Redirects.RedirectsInitializer.Initialize()
   at Geta.NotFoundHandler.Infrastructure.Initialization.ApplicationBuilderExtensions.UseNotFoundHandler(IApplicationBuilder app)

**Comment by lars-w-pedersen** [comment]
i get this in the log file:

2021-07-28 06:37:41.370 +02:00 [ERR] Error while running stored procedure.
2021-07-28 06:37:41.370 +02:00 [INF] Create NotFoundHandler redirects table START
2021-07-28 06:37:41.371 +02:00 [ERR] An error occurred in the ExecuteSQL method with the following sql: CREATE TABLE [dbo].[NotFoundHandler.Redirects] (
                                        [Id] [uniqueidentifier] NOT NULL,
                                        [OldUrl] [nvarchar](2000) NOT NULL,
                                        [NewUrl] [nvarchar](2000) NOT NULL,
                                        [State] [int] NOT NULL,
                                        [WildCardSkipAppend] [bit] NOT NULL,
                                        [RedirectType] [int] NOT NULL,
                                        CONSTRAINT [PK_NotFoundHandlerRedirects] PRIMARY KEY CLUSTERED ([Id] ASC) ON [PRIMARY]
                                        ) ON [PRIMARY]
2021-07-28 06:37:41.371 +02:00 [INF] Create NotFoundHandler redirects table END
2021-07-28 06:37:41.380 +02:00 [ERR] An error occurred in the ExecuteSQL method with the following sql: SELECT Id, OldUrl, NewUrl, State, WildCardSkipAppend, RedirectType FROM [dbo].[NotFoundHandler.Redirects]

**Comment by lars-w-pedersen** [comment]
The problem is in this code in ServiceCollectionExtensions 

        public static IServiceCollection AddNotFoundHandler(
            this IServiceCollection services,
            Action<NotFoundHandlerOptions> setupAction)
        {
            return AddNotFoundHandler(services, o => { }, DefaultPolicy);
        }

where setupAction is ignored

**Comment by lars-w-pedersen** [comment]
by the way this line:
_logger.LogError($"An error occurred in the ExecuteSQL method with the following sql: {sqlCommand}", ex);
do not log the ex error. It should be:
_logger.LogError(ex, $"An error occurred in the ExecuteSQL method with the following sql: {sqlCommand}");
the same goes for all the other places where errors are logged

**Comment by marisks** [comment]
Thank you @lars-w-pedersen ! 

It was quite unusual to put exception as a first parameter. Previously, with log4net it was the opposite.

**Comment by usamamalik75** [comment]
@lars-w-pedersen I was facing the same issue 
System.IndexOutOfRangeException
HResult=0x80131508
Message=Cannot find table 0.

**But I am able to fix this issue using this line of codes**

**in ConfigureServices:**
```
services.AddNotFoundHandler(o =>
            {
                o.UseSqlServer(_configuration.GetConnectionString("EPiServerDB"));
                o.AddProvider<NullNotFoundHandlerProvider>();
            }, policy =>
            {
                policy.RequireRole(Roles.CmsAdmins);
            });
            services.AddOptimizelyNotFoundHandler();
```
            
**in Configure:**
```
app.UseNotFoundHandler();
```

**position of the code should be at the end not in start.**

![image](https://user-images.githubusercontent.com/33087810/143500165-a17d9c30-290d-4cac-95bd-2de0663390e8.png)

**Comment by marisks** [comment]
@usamamalik75 Do you have an issue with the latest version (1.0.1)? It should be fixed.

**Comment by usamamalik75** [comment]
@marisks I am using the Geta.NotFoundHandler.Optimizely which has stable version 1.0.0.

**Comment by marisks** [comment]
@usamamalik75 The 1.0.1 is also the stable version with the bugfix. You should use it.

**Comment by marisks** [comment]
Closed as fixed in 1.0.1

**Comment by cjsharp1** [comment]
Hey @marisks! I'm still looking more into this, but I see this issue in 1.0.4, only when trying to view the NotFoundHandler Optimizely Admin UI.
